WordPress/Sniffs/Theme/NoAutoGenerateSniff.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for auto generated themes.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoAutoGenerateSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$tokens   = PHP_CodeSniffer_Tokens::$stringTokens;
+		$tokens[] = T_INLINE_HTML;
+		$tokens[] = T_COMMENT;
+		return $tokens;
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		// No need to check an empty string.
+		if ( '' === trim( $token['content'] ) ) {
+			return;
+		}
+
+		$generated_themes = array(
+			// Artisteer.
+			'artisteer',
+			// Lubith.
+			'lubith',
+			// TemplateToaster.
+			'templatetoaster',
+			// WP Theme Generator.
+			'wpthemegenerator',
```

Comment:
Ah, I misread that comment before.

Still, serial theme builders often don't download a new version of base code for each theme (take TGMPA for example... sigh...), so should we may be still check for code from at least a few versions back ?